File: 14-Python/hello_world/classes.py
class Person:
    def __init__(self, first_name, last_name):
        self.first_name = first_name
        self.last_name = last_name

    def __str__(self):
        return "%(first_name)s %(last_name)s" % {
            "first_name": self.first_name,
            "last_name": self.last_name
        }


class Student(Person):
    def __init__(self, first_name, last_name, grade):
        super().__init__(first_name, last_name)
        self.grade = grade


students = [Student("Jonas", "Jonaitis", 5),
            Student("Petras", "Petraitis", 8),
            Student(last_name="Kirvaitis", first_name="Aštrys",
                    grade=7)]

best_student = max(students, key=lambda student: student.grade)
print(best_student)

File: 15-WebWorkers/language-examples/classes.js
class Person {
    constructor(firstName, lastName) {
        this.firstName = firstName;
        this.lastName = lastName;
    }

    toString() {
        return `${this.firstName} ${this.lastName}`;
    }
}

class Student extends Person {
    constructor(firstName, lastName, grade) {
        super(firstName, lastName);
        this.grade = grade;
    }
}

const students = [
    new Student("Jonas", "Jonaitis", 5),
    new Student("Petras", "Petraitis", 8),
    new Student("Aštrys", "Kirvaitis", 7)
];
const bestStudent = students
    .reduce((best, current) => current.grade > best.grade ? current : best);
console.log(bestStudent.toString());
